### PR TITLE
Support compound indexes generation

### PR DESCRIPTION
**Pull request by Hexilee**

### CODE REVIEW

File: cmd/index.go [hunk 1]
```diff
@@ -142,3 +170,83 @@ func pathExist(path string) bool {
 	_, err := os.Stat(path)
 	return err == nil || !os.IsNotExist(err)
 }
+
+func keyName(table *types.Table, fields []string) string {
+	segments := append([]string{strings.ToUpper(table.Name.String())}, fields...)
+	segments = append(segments, "ID")
```
Comment (mahjonp) [inline cmd/index.go:176]:
```suggestion
	segments = append(segments, "IDX")
```

File: cmd/index.go [hunk 2]
```diff
@@ -142,3 +170,83 @@ func pathExist(path string) bool {
...
+	if len(allPairs) <= max {
+		return allPairs
+	}
+	pairMap := make(map[IndexDMLPair]bool)
+	for len(pairMap) < max {
+		pair := allPairs[generator.Rd(len(allPairs))]
+		pairMap[pair] = true
+	}
+
+	ret := allPairs[:0]
```
Comment (mahjonp) [inline cmd/index.go:238]:
```suggestion
	ret := make([]IndexDMLPair, 0)
```

File: cmd/index.go [hunk 3]
```diff
@@ -142,3 +170,83 @@ func pathExist(path string) bool {
...
+		fields := []string{column.Name.String()}
+		fieldLists = append(fieldLists, fields)
+		if column.Key == "" {
+			pairs = append(pairs, fields2Pair(table, fields))
+		}
+	}
+	allLevelPairs = append(allLevelPairs, pairs)
+	allLevelFieldLists = append(allLevelFieldLists, fieldLists)
+
+	for i := 1; i < level; i++ {
```
Comment (mahjonp) [inline cmd/index.go:203]: i <= level?

File: cmd/index.go [hunk 4]
```diff
@@ -142,3 +170,83 @@ func pathExist(path string) bool {
...
+			}
+			for _, column := range table.Columns {
+				if !fieldSet[column.Name.String()] {
+					newList := append(list, column.Name.String())
+					fieldLists = append(fieldLists, newList)
+					pairs = append(pairs, fields2Pair(table, newList))
+				}
+			}
+			for _, field := range list {
+				delete(fieldSet, field)
```
Comment (mahjonp) [inline cmd/index.go:219]: Why not empty fieldSet like `fieldSet = make(map[string]bool)`?

File: cmd/index.go (hunk 3 above)
Comment (Hexilee) [reply cmd/index.go:203]: It does be `i < level` as the index of slice starts from 0.

File: cmd/index.go (hunk 4 above)
Comment (Hexilee) [reply cmd/index.go:219]: For reuse memory

Comment (Hexilee) [reply cmd/index.go:219]: Seems `fieldSet = make(map[string]bool)` is more effective, updated

File: cmd/index.go (hunk 2 above)
Comment (Hexilee) [reply cmd/index.go:238]: Resolved

File: cmd/index.go (hunk 1 above)
Comment (Hexilee) [reply cmd/index.go:176]: Resolved

**Pull request merged**